components/backup/src/endpoint.rs

```diff
@@ -568,10 +568,10 @@ fn get_max_start_key(start_key: Option<&Key>, region: &Region) -> Option<Key> {
 
 /// Construct an backup file name based on the given store id and region.
 /// A name consists with three parts: store id, region_id and a epoch version.
-fn backup_file_name(store_id: u64, region: &Region, key: Option<String>) -> String {
+fn backup_file_name(store_id: u64, region: &Region, key: Option<Vec<u8>>) -> String {
     match key {
         Some(k) => format!(
-            "{}_{}_{}_{}",
+            "{}_{}_{}_{:?}",
             store_id,
             region.get_id(),
             region.get_region_epoch().get_version(),
```

Comment:
Should use `"{}", hex::encode(k)`.

Here is an example of `{:?}, vec![1, 2, 3]` => `[1, 2, 3] `

Reply:
`hex::encode` is used in [sha256](https://github.com/tikv/tikv/blob/64b905954c1f4fdf00c10e3f64932240b0843d13/components/tikv_util/src/file.rs#L85, I think use `String::from_utf8` is enough